File: sender-android/app/src/test/java/com/openaudiolink/TestFixtures.kt
package com.openaudiolink

import java.io.File
import java.io.FileNotFoundException

internal object TestFixtures {
    fun read(relativePath: String): ByteArray {
        var directory: File? = File(System.getProperty("user.dir") ?: ".").absoluteFile
        while (directory != null) {
            val candidate = File(directory, relativePath)
            if (candidate.isFile) return candidate.readBytes()
            directory = directory.parentFile
        }
        throw FileNotFoundException("Fixture not found: $relativePath")
    }
}

File: receiver-windows/tests/OpenAudioLink.Tests/TestFixtures.cs
using System;
using System.IO;

namespace OpenAudioLink.Tests
{
    internal static class TestFixtures
    {
        public static byte[] Read(string relativePath)
        {
            DirectoryInfo directory = new DirectoryInfo(AppDomain.CurrentDomain.BaseDirectory);
            while (directory != null)
            {
                string path = Path.Combine(directory.FullName, relativePath);
                if (File.Exists(path))
                {
                    return File.ReadAllBytes(path);
                }

                directory = directory.Parent;
            }

            throw new FileNotFoundException("Fixture not found.", relativePath);
        }
    }
}
